### PR TITLE
Feature/gh 2651

### LINKED ISSUE #2651: Add config 'enabled' in personalization
personalization can be enabled in backend. In our current SPA, it always supports `personalization` (adding personalization request headers) even it is disabled in backend.

So, only when `enabled=true`, SPA supports personalization. If `enabled=false`, the personalization related data saved in local storage should be cleared.

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #2651

### CODE REVIEW

File: projects/core/src/personalization/http-interceptors/occ-personalization-id.interceptor.ts [hunk 1]
```diff
@@ -20,24 +20,31 @@ const PERSONALIZATION_ID_KEY = 'personalization-id';
...
     private config: PersonalizationConfig,
     private occEndpoints: OccEndpointsService,
     private winRef: WindowRef,
     @Inject(PLATFORM_ID) private platform: any
   ) {
-    this.requestHeader = this.config.personalization.httpHeaderName.id.toLowerCase();
-    this.personalizationId =
-      this.winRef.localStorage &&
-      this.winRef.localStorage.getItem(PERSONALIZATION_ID_KEY);
+    this.enabled = this.config.personalization.enabled || false;
```
Comment (plabadie) [inline projects/core/src/personalization/http-interceptors/occ-personalization-id.interceptor.ts]: Why `|| false`?

File: projects/core/src/personalization/http-interceptors/occ-personalization-time.interceptor.spec.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import { WindowRef } from '../../window/window-ref';
 
 const mockPersonalizationConfig: PersonalizationConfig = {
   personalization: {
+    enabled: true,
```
Comment (plabadie) [inline projects/core/src/personalization/http-interceptors/occ-personalization-time.interceptor.spec.ts]: Please add coverage for what to expect when enabled == false.

File: projects/core/src/personalization/http-interceptors/occ-personalization-id.interceptor.ts [hunk 3]
```diff
@@ -20,24 +20,31 @@ const PERSONALIZATION_ID_KEY = 'personalization-id';
 export class OccPersonalizationIdInterceptor implements HttpInterceptor {
   private personalizationId: string;
   private requestHeader: string;
+  private enabled: boolean;
```
Comment (plabadie) [inline projects/core/src/personalization/http-interceptors/occ-personalization-id.interceptor.ts]: Please add coverage for what to expect when enabled == false.

File: projects/core/src/personalization/http-interceptors/occ-personalization-id.interceptor.ts (hunk 1 above)
Comment (WeizhengSap) [reply projects/core/src/personalization/http-interceptors/occ-personalization-id.interceptor.ts]: If there is no value in `this.config.personalization.enabled`, we want `this.enabled` is `false`.

**Pull request merged**